pages/products/[id].tsx

```diff
@@ -0,0 +1,224 @@
+import React, { useState, MouseEvent, ChangeEvent } from 'react';
+import Head from 'next/head';
+import Nav from '../../components/Nav';
+import Footer from '../../components/Footer';
+import Cookies from 'js-cookie';
+import Link from 'next/link';
+import { NextPageContext } from 'next';
+
+// using typescript: 	yarn add --dev typescript @types/node
+
+type Items = {
+  id: string;
+  name: string;
+  price: number;
+  src: string;
+  h2: string;
+  h3: string;
+  p: string;
+};
+type Props = { items: Items };
+
+// [id] => act as the part of the Path like 'localhost:3000/users/1(which is [id]'
+const Product = (props: Props) => {
+  const [price, setPrice] = useState<number>();
+  // typescript
+  // const [piece, setPiece] = useState<number | undefined>(undefined);
+  const [piece, setPiece] = useState(1);
+
+  // typescript error: Parameter 'e' implicitly has an 'any' type
+  function changePieces(e: ChangeEvent<HTMLInputElement>) {
+    setPiece(Number(e.target.value));
+  }
+
+  // after line 31, type of piece is updated into number
+  function showPrice() {
+    setPrice(piece * props.items.price);
+  }
+
+  if (!props.items) return <div>Item not found!</div>;
+
+  function goCart() {
+    const product = {
+      piece: piece,
+      price: price,
+      name: props.items.name,
+      src: props.items.src,
+      id: props.items.id,
+    };
+
+    const newCart = Cookies.getJSON('cart') || [];
+
+    newCart.push(product);
+
+    Cookies.set('cart', newCart);
+    window.location.reload();
+  }
+
+  return (
+    <div>
+      <Head>
+        <title>{props.items.name}</title>
+        <link rel="icon" href="/favicon.png" />
+      </Head>
+      <Nav />
+
+      <main>
+        <h1>{props.items.h2}</h1>
+        <h2>{props.items.name}</h2>
+        <img src={props.items.src} alt="items" />
+        <h3>{props.items.h3}</h3>
+        <p>{props.items.p}</p>
+
+        <p>Euro: {props.items.price}</p>
+        <hr />
+
+        <form>
+          <input
+            type="text"
+            min="1"
+            step="10"
+            placeholder="pieces"
+            onChange={changePieces}
+            // without called showPrice, got this error: Type '(event: FormEvent<HTMLInputElement>) => void' is not assignable to type '() => any'
+            onKeyUp={showPrice}
+            value={piece}
+          />
+        </form>
+        <br />
+        <hr />
+
+        <p>Qty: {piece} </p>
+        <p>Total Price: {price} </p>
+        <hr />
+
+        <button data-cy="addCart-button" onClick={() => goCart()}>
+          Add items
+        </button>
+
+        {/* Use Typescript, for Link error add : yarn upgrade @types/react@latest  */}
+        <Link href="/cartForPayment">
+          <a>
+            <button data-cy="go-to-cart-button" className="toCartButton">
```

Comment:
I guess having a button inside of the `Link` is probably not a great idea (a `Link` generates an `a` tag, and having a `button` inside of that can create unexpected situations).

If you need a `data-cy` and `className`, then you can put this on the `Link` itself, I think.